x-pack/plugins/ml/public/application/explorer/dashboard_controls/add_anomaly_charts_to_dashboard_controls.tsx

```diff
@@ -6,33 +6,33 @@
  */
 import React, { FC, useCallback, useState } from 'react';
 import { FormattedMessage } from '@kbn/i18n-react';
-import { EuiFieldNumber, EuiFormRow, formatDate, htmlIdGenerator } from '@elastic/eui';
-import { TimeRange } from '@kbn/data-plugin/common/query';
+import { EuiFieldNumber, EuiFormRow, htmlIdGenerator } from '@elastic/eui';
+import { Query } from '@kbn/es-query';
+import { escapeKueryForFieldValuePair } from '../../util/string_utils';
+import { SEARCH_QUERY_LANGUAGE } from '../../../../common/constants/search';
 import { useDashboardTable } from './use_dashboards_table';
 import { AddToDashboardControl } from './add_to_dashboard_controls';
 import { useAddToDashboardActions } from './use_add_to_dashboard_actions';
-import { AppStateSelectedCells, getSelectionTimeRange } from '../explorer_utils';
+import { AppStateSelectedCells, getSelectionInfluencers } from '../explorer_utils';
 import { DEFAULT_MAX_SERIES_TO_PLOT } from '../../services/anomaly_explorer_charts_service';
 import { JobId } from '../../../../common/types/anomaly_detection_jobs';
 import { ANOMALY_EXPLORER_CHARTS_EMBEDDABLE_TYPE } from '../../../embeddables';
 import { getDefaultExplorerChartsPanelTitle } from '../../../embeddables/anomaly_charts/anomaly_charts_embeddable';
-import { TimeRangeBounds } from '../../util/time_buckets';
 import { useTableSeverity } from '../../components/controls/select_severity';
 import { MAX_ANOMALY_CHARTS_ALLOWED } from '../../../embeddables/anomaly_charts/anomaly_charts_initializer';
 
-function getDefaultEmbeddablePanelConfig(jobIds: JobId[]) {
+function getDefaultEmbeddablePanelConfig(jobIds: JobId[], queryString?: string) {
   return {
     id: htmlIdGenerator()(),
-    title: getDefaultExplorerChartsPanelTitle(jobIds),
+    title: getDefaultExplorerChartsPanelTitle(jobIds).concat(queryString ? `- ${queryString}` : ''),
```

Comment:
I reckon the query might be too cumbersome as part of the title. I think for the dashbard we can keep it as is. 
As for the case's attachment, we can add the query on top of the embeddable input content [here](https://github.com/elastic/kibana/blob/cb7b03106fd6b1f6560ad9a5cc954e59095ea700/x-pack/plugins/ml/public/cases/anomaly_charts_attachments.tsx#L37)

Reply:
That makes sense, what do you think about showing the applied query and filters at the top of the embeddable panel?

Reply:
@darnautov Like how we show it in case embeddable? That would be a great idea. Although I'm not sure how much more benefit it brings vs to having it to be in the title so that users can remove/modify the title if they want to as supposed to taking screen real estate of the chart without a way to remove. Unless Dashboard has a baked in way of showing that the filter/query for the panel is customized like how it does for the time range?